src/daemons/rhsmcertd.c

```diff
@@ -256,24 +256,22 @@ cert_check (gboolean heal)
 struct CertCheckData {
     int cert_interval_seconds;
     int heal_interval_seconds;
-    bool heal;
 };
 
 static gboolean
 initial_cert_check (gpointer data)
 {
     struct CertCheckData *cert_data = data;
-    cert_check (cert_data->heal);
+    cert_check (true);  // Do the initial auto attach
+    cert_check (false);  // Do the initial cert check (different than above)
+    bool heal = true;
     // Add the timeout to begin waiting on interval but offset by the initial
     // delay.
-    if (cert_data->heal) {
-        g_timeout_add (cert_data->heal_interval_seconds * 1000,
-            (GSourceFunc) cert_check, (gpointer) cert_data->heal);
-    }
-    else {
-        g_timeout_add (cert_data->cert_interval_seconds * 1000,
-            (GSourceFunc) cert_check, (gpointer) cert_data->heal);
-    };
+    g_timeout_add (cert_data->heal_interval_seconds * 1000,
+        (GSourceFunc) cert_check, (gpointer) heal);
```

Comment:
I noticed we're doing some weird stuff here with casting `gpointer` to `bool` and vice-versa. I added a card to the backlog for this. It's fairly hacky. But this PR doesn't make us any worse than the status quo (been this way for a long time), so I think we should address it later.